pkg/templates/funcs_test.go

```diff
@@ -0,0 +1,59 @@
+package templates
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFromYAML(t *testing.T) {
+	tests := []struct {
+		name    string
+		yaml    string
+		want    interface{}
+		wantErr bool
+	}{
+		{
+			"invalid yaml",
+			"'",
+			nil,
+			true,
+		},
+		{
+			"empty object",
+			"{}",
+			map[interface{}]interface{}{},
+			false,
+		},
+		{
+			"raw string",
+			"hello world",
+			"hello world",
+			false,
+		},
+		{
+			"array of strings",
+			`["hello", "goodbye"]`,
+			[]interface{}{"hello", "goodbye"},
+			false,
+		},
+		{
+			"array of objects",
+			`["hello", {}]`,
+			[]interface{}{"hello", map[interface{}]interface{}{}},
```

Comment:
nit: i assume this doesnt care what is inside the yaml, and should work any validd yaml like parsing key value pairs 
````
age: "23"
hello: bar
```

Reply:
That's right, added a test to demonstrate.